pandas/core/arrays/base.py

```diff
@@ -351,6 +351,17 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def __contains__(self, item) -> bool:
+        """
+        Return for `item in self`.
+        """
+        # comparisons of any item to pd.NA always return pd.NA, so e.g. "a" in [pd.NA]
+        # would raise a TypeError. The implementation below works around that.
+        if isna(item):
```

Comment:
is_valid_nat_for_dtype

Reply:
I don't understand you here. Are you saying that `pd.NaT in arr` should return False if the array is not a datetime-like array? Can you expand a bit?

Reply:
> Are you saying that pd.NaT in arr should return False if the array is not a datetime-like array? 

correct (unless maybe object dtype).  Some other examples:

```
dta = pd.date_range("2016-01-01", periods=3)._data  # <-- DateTimeArray
dta[-1] = pd.NaT
pa = dta.to_period("D")  # <-- PeriodArray
tda = dta - dta  # <-- TimedeltaArray
arr = pd.array([1, 2 ,3, pd.NA])  # <-- IntegerArray
flt = pd.array([1.0, 2.0, 3.0, np.nan])  # <-- FloatArray

tdnat = np.timedelta64("NaT", "ns")
dtnat = np.timedelta64("NaT", "ns")

>>> tdnat in dta
False
>>> dtnat in dta
True  # <-- actually this returns False, but thats a newly-discovered bug

>>> tdnat in pa
False
>>> dtnat in pa
False

>>> tdnat in tda
True   # <-- actually this raises TypeError, but thats a newly-discovered bug
>>> dtnat in tda
False    # <-- actually this raises TypeError, but thats a newly-discovered bug

>>> tdnat in flt
False
>>> dtnat in flt
False    # <-- actually this raises TypeError, but thats a newly-discovered bug

>>> tdnat in arr
False
>>> dtnat in arr
False     # <-- actually this raises TypeError, but thats a newly-discovered bug
```

side-note after checking all of these: yikes!

Reply:
I don't use datetime-likes very much, so I'm a bit weak on what these nan-like comparisons to nan-likes should return in datatime-likes and you certainly didn't help there :-)

Maybe do as @jreback suggests, and handle these later/separately? An alternative could be in the base `Expensionarray` check for `pd.NA` only. It sub-classes wan to do it differently, they can implement a `__contains__` method themselves. That seems the cleanest to me.